### PR TITLE
Add apikey auth

### PR DESCRIPTION
**Pull request by An-Li-magicloud**
* [X] API key auth
* [X] tidy3d cli

### CODE REVIEW

File: tests/test_web/test_cli.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+from os.path import expanduser
+
+from click.testing import CliRunner
+
+from tidy3d.web.cli import tidy3d_cli
+
+
+def test_tidy3d_cli():
+    home = expanduser("~")
+    if os.path.exists(f"{home}/.tidy3d/config"):
```
Comment (tylerflex) [inline tests/test_web/test_cli.py:12]: best to import these paths from `tidy3d.web.cli` in case they change later.

**Review by tylerflex — CHANGES_REQUESTED** [review]
just a minor comment about using the config file path defined in cli.py otherwise looks good. I tested a few scenarios out in this branch and they all worked as expected.
* using old authentication (auth.json file) worked as expected, no prompting for API key.
* remove auth.json file, prompted for email and password as before.
* `tidy3d configure` with wrong API key gives helpful message and exits.
* `tidy3d configure` with correct API key works as expected.

**Review by tylerflex — APPROVED** [review]

**Comment by tylerflex** [comment]
fixes #657

**Review by tylerflex — CHANGES_REQUESTED** [review]
Hi An, I just tested this and there are a few issues still:
1. When I put in the correct API key, I get the following message:
```
Configured successfully.
API key is invalid.
```
we should fix it so it doesn't say that the API key is invalid if it is configured successfully.

2. the prompt formatting is a bit strange, this is what I see (for old key of '{key}')
```
API Key:
Current:[{key}
New:]: 
```
I feel like it should problbably rather be
```
API Key:
Current: [{key}]
New:
```
3. I still don't know if Lei's comments were addressed [here](https://github.com/flexcompute/tidy3d/issues/657). I think his point was that:
 * if `tidy3d configure` is called, we want to prompt for username / password, grab the API key and store that locally. 
 * If `--api-key` is used, we enter the username / API key directly.
right now I think the `api-key` is just used to enter it in the command like
```
tidy3d configure --api-key={key}
```
? if so, this is fine but maybe eventually we may want to change the default behavior of `tidy3d configure` to prompt for email / password and store the API key locally. Does that makes sense?

**Review by tylerflex — APPROVED** [review]

**Comment by tylerflex** [comment]
I think this is almost ready to merge into 1.9. @dbochkov-flexcompute if you don't mind, could you take a look and see if you notice any potential issues that I might have missed? thanks

File: README.md [hunk 2]
```diff
@@ -63,6 +97,14 @@ and it should print out the version number, for example:
 1.0.0
 ```
 
+#### Authentication
+
+To test the web / authentication
+
+```
+python -c "import tidy3d.web"
```
Comment (dbochkov-flexcompute) [inline README.md:105]: Is this command supposed to test authentication? I still get the same response 
```
[11:25:02] INFO     Using client version: 1.9.0  
```
with and without a valid authentication file

**Review by dbochkov-flexcompute — COMMENTED** [review]
Everything looks good to me to the extent I can judge. Just note that I can still reproduce that weird error when having a valid `auth.json` and an invalid `config` in the following way:
- start with having a valid `auth.json` 
- execute `tidy3d configure` and provide a correct API key
- go to tidy3d website, press `New API key`, and "forget" to reconfigure tidy3d

File: README.md (hunk 2 above)
Comment (tylerflex) [reply README.md:105]: I think this readme is just out of date, sorry. I usually do something like
```py
python -c "import tidy3d.web as web; web.get_info(0)
```
just to test whether the authentication completed successfully.

**Comment by tylerflex** [comment]
> Everything looks good to me to the extent I can judge. Just note that I can still reproduce that weird error when having a valid `auth.json` and an invalid `config` in the following way:
> 
> * start with having a valid `auth.json`
> * execute `tidy3d configure` and provide a correct API key
> * go to tidy3d website, press `New API key`, and "forget" to reconfigure tidy3d

@An-Li-magicloud Think you could fix this? should be the last thing. Thanks

**Comment by An-Li-magicloud** [comment]
> > Everything looks good to me to the extent I can judge. Just note that I can still reproduce that weird error when having a valid `auth.json` and an invalid `config` in the following way:
> > 
> > * start with having a valid `auth.json`
> > * execute `tidy3d configure` and provide a correct API key
> > * go to tidy3d website, press `New API key`, and "forget" to reconfigure tidy3d
> 
> @An-Li-magicloud Think you could fix this? should be the last thing. Thanks

This is due to retry logic, it hard to detect which auth way cause the error. I would recommend remove the retry logic.

``` python
        while resp.status_code == ResponseCodes.UNAUTHORIZED.value and attempts < MAX_ATTEMPTS:
            # ask for credentials and call the http request again
            get_credentials()
            resp = func(*args, **kwargs)
            attempts += 1
```

**Comment by tylerflex** [comment]
> This is due to retry logic, it hard to detect which auth way cause the error. I would recommend remove the retry logic.
I'm ok with removing it. lets do that

**Pull request merged**